OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/AndroidICCAnalysis.scala

```diff
@@ -658,58 +603,61 @@ class AndroidICCAnalysis(val project: SomeProject) extends FPCFAnalysis {
+                    val addUseSites = na.targetVar.usedBy.toChain.toList
+                    addUseSites.foreach { i ⇒
+                        val s = stmts(i)
+                        if (s.isNonVirtualMethodCall && s.asNonVirtualMethodCall.name == init && c.pc != s.pc) {
+                            reconstructIntent(s.asNonVirtualMethodCall, tacCode, useSites ++ addUseSites, m)
                         }
-                    } else {
-                        //intent comes from another method => overapproximation
-                        val intent = new ImplicitIntent(m)
-                        intent.iData = undef
-                        intent.action = undef
-                        intent.categories += undef
-                        evaluateUseSites(tacCode, useSites, intent)
                     }
+                } else {
+                    //intent comes from another method => overapproximation
+                    val intent = new ImplicitIntent(m)
+                    intent.iData = undef
+                    intent.action = undef
+                    intent.categories += undef
+                    evaluateUseSites(tacCode, useSites, intent)
                 }
-            case 2 ⇒ //Intent(String action, Uri uri)
+            case MethodDescriptor(types, _) if types == FieldTypes(FieldType("Ljava/lang/String;"), FieldType("Landroid/content/Intent;")) ⇒ //Intent(String action, Uri uri)
                 if (p.head.asVar.value.verificationTypeInfo == ObjectVariableInfo(ObjectType.String)) {
 
                     val intent = new ImplicitIntent(m)
                     intent.action = p.head.asVar.value.toString.replaceAll(paramRegex, "")
                     intent.iData = (checkData(1, tacCode, p))
                     evaluateUseSites(tacCode, useSites, intent)
 
-                } else { //Intent(Context context, Class cls)
-                    val clsFile = project.classFile(ObjectType(p.last.asVar.value.toString.
-                        replaceAll(valueRegex, "").replaceAll("\\.", "/")))
+                }
+            case MethodDescriptor(types, _) if types == FieldTypes(FieldType("Landroid/content/Context;"), FieldType("Ljava/lang/Class;")) ⇒
+                //Intent(Context context, Class cls)
+                val clsFile = project.classFile(ObjectType(p.last.asVar.value.toString.
+                    replaceAll(valueRegex, "").replaceAll("\\.", "/")))
 
-                    if (clsFile.isDefined) {
-                        val intent = new ExplicitIntent(m, clsFile.get)
-                        intent.findStartedMethod(tacCode, useSites)
-                        if (intent.calledMethod.nonEmpty) {
-                            explicitIntents += intent
-                        }
+                if (clsFile.isDefined) {
+                    val intent = new ExplicitIntent(m, clsFile.get)
+                    intent.findStartedMethod(tacCode, useSites)
```

Comment:
This seems like a bad pattern (calling a mutating method on the Intent object. Is that startedMethod always needed anyway? Then it should probably be initialized by the constructor of the Intent object.